packages/wdio-utils/src/envDetector.ts

```diff
@@ -190,6 +200,44 @@ function isAndroid(capabilities?: WebdriverIO.Capabilities) {
     )
 }
 
+/**
+ * Check if session uses a specific automation name
+ * @param  {Object}  capabilities  caps of session response
+ * @param  {String}  platform      platform to check for (e.g., 'windows', 'mac2')
+ * @return {Boolean}               true if run for specified platform
+ */
+function matchesAppAutomationName(automationNameValue: string, capabilities?: WebdriverIO.Capabilities): boolean {
+    if (!capabilities) {
+        return false
+    }
+
+    const automationName = getAutomationName(capabilities)
+
+    if (!automationName) {
+        return false
+    }
+
+    return Boolean(automationName.match(new RegExp(automationNameValue, 'i')))
```

Comment:
```suggestion
    return new RegExp(automationNameValue, 'i').test(automationName)
```

Reply:
@erwinheitzman 

This would return an array and match a boolean, or did I learn this wrong last time?